### PR TITLE
Add DisableImplicit1xNuGetFallbackFolder flag

### PR DESCRIPTION
**Pull request by wli3**
fix https://github.com/dotnet/cli/issues/9563

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.NuGetOfflineCache.targets [hunk 1]
```diff
@@ -22,8 +22,8 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <PropertyGroup Condition=" '$(DisableImplicitNuGetFallbackFolder)' != 'true' and Exists($(_NugetFallbackFolder)) ">
     <RestoreAdditionalProjectSources Condition=" '$(_IsNETCore1x)' == 'true' ">$(RestoreAdditionalProjectSources);$(_NugetFallbackFolder)</RestoreAdditionalProjectSources>
-    <RestoreAdditionalProjectFallbackFoldersExcludes Condition=" '$(_IsNETCore1x)' == 'true' ">$(RestoreAdditionalProjectFallbackFoldersExcludes);$(_NugetFallbackFolder)</RestoreAdditionalProjectFallbackFoldersExcludes>
+    <RestoreAdditionalProjectFallbackFoldersExcludes Condition=" '$(_IsNETCore1x)' == 'true' and '$(DisableImplicit1xNuGetFallbackFolder)' != 'true' ">$(RestoreAdditionalProjectFallbackFoldersExcludes);$(_NugetFallbackFolder)</RestoreAdditionalProjectFallbackFoldersExcludes>
```
[the remark this answers is not in the document]
Comment (livarcocc) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.NuGetOfflineCache.targets]: Seems good.

File: src/Tests/Microsoft.NET.Restore.Tests/GivenThatWeWantToRestoreProjectsUsingNuGetConfigProperties.cs [hunk 2]
```diff
@@ -71,6 +71,30 @@ public void I_can_disable_implicit_msbuild_nuget_config(string frameworks, strin
             restoreCommand.Execute($"/p:DisableImplicitNuGetFallbackFolder=true").Should().Fail();
         }
 
+        [Theory]
+        [InlineData("netstandard1.3", "1.3", true)]
+        [InlineData("netcoreapp1.0", "1.0", false)]
+        [InlineData("netcoreapp1.1", "1.1", false)]
+        [InlineData("netstandard2.0", "2.0", true)]
+        [InlineData("netcoreapp2.0", "2.0app", true)]
+        public void I_can_disable_1_x_implicit_msbuild_nuget_config(string frameworks, string projectPrefix, bool shouldExecutePass)
```
[the remark this answers is not in the document]
Comment (livarcocc) [reply src/Tests/Microsoft.NET.Restore.Tests/GivenThatWeWantToRestoreProjectsUsingNuGetConfigProperties.cs]: Also seems good.

**Review by livarcocc — APPROVED** [review]

**Review by JunTaoLuo — CHANGES_REQUESTED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.NuGetOfflineCache.targets [hunk 3]
```diff
@@ -22,7 +22,7 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <PropertyGroup Condition=" '$(DisableImplicitNuGetFallbackFolder)' != 'true' and Exists($(_NugetFallbackFolder)) ">
     <RestoreAdditionalProjectSources Condition=" '$(_IsNETCore1x)' == 'true' ">$(RestoreAdditionalProjectSources);$(_NugetFallbackFolder)</RestoreAdditionalProjectSources>
```
[the remark this answers is not in the document]
Comment (livarcocc) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.NuGetOfflineCache.targets]: You are right, this is what needs to change. My bad for not catching this.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.NuGetOfflineCache.targets (hunk 1 above)
[the remark this answers is not in the document]
Comment (livarcocc) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.NuGetOfflineCache.targets]: @JunTaoLuo you are right. I missed it when I originally reviewed this.

File: src/Tests/Microsoft.NET.Restore.Tests/GivenThatWeWantToRestoreProjectsUsingNuGetConfigProperties.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (livarcocc) [reply src/Tests/Microsoft.NET.Restore.Tests/GivenThatWeWantToRestoreProjectsUsingNuGetConfigProperties.cs]: Given that our implementation above was wrong and this test didn't catch this, I wonder if this coverage is right.

[the remark this answers is not in the document]
Comment (wli3) [reply src/Tests/Microsoft.NET.Restore.Tests/GivenThatWeWantToRestoreProjectsUsingNuGetConfigProperties.cs]: well, it is failing

**Review by JunTaoLuo — COMMENTED** [review]
The implementation now looks correct but the tests are failing. Maybe the restore you expect to fail are passing because of caching or you didn't disable online sources (nuget, etc).

**Comment by wli3** [comment]
@JunTaoLuo I am trying to verify these tests cannot run locally... which apparently true. (all tests in the file are red in test explore) so sorry for the noise.

You can see the test above it, the disabling all NuGetFallbackFolder is doing the same thing to assert fail. Cached package may not be the problem. Let me keep digging

**Comment by wli3** [comment]
@JunTaoLuo you are right. I added extra code to avoid cache being used. It is good now?

**Review by JunTaoLuo — APPROVED** [review]
Yup looks good

**Pull request merged**